debian/patches/revert-e30549e8-remove-networkd-online-dependency.patch

```diff
@@ -0,0 +1,821 @@
+Description: Revert e30549e8 - remove networkd online dependency
+ This change can cause failures on instances that have changes
+ to the default systemd service ordering or a complex network
+ setup. This also implicitly reverts 65e9788 as that fixes
+ some upstream functionality in e30549e8.
+Author: James Falcon <james.falcon@canonical.com>
+Bug: https://bugs.launchpad.net/ubuntu/+source/cloud-init/+bug/2094149
+Last-Update: 2025-01-15
+--- a/cloudinit/cmd/main.py
++++ b/cloudinit/cmd/main.py
+@@ -19,7 +19,7 @@ import sys
+ import traceback
+ import logging
+ import yaml
+-from typing import Optional, Tuple, Callable, Union
++from typing import Tuple, Callable
+ 
+ from cloudinit import netinfo
+ from cloudinit import signal_handler
+@@ -34,13 +34,11 @@ from cloudinit import warnings
+ from cloudinit import reporting
+ from cloudinit import atomic_helper
+ from cloudinit import lifecycle
+-from cloudinit import handlers
+ from cloudinit.log import log_util, loggers
+ from cloudinit.cmd.devel import read_cfg_paths
+ from cloudinit.config import cc_set_hostname
+ from cloudinit.config.modules import Modules
+ from cloudinit.config.schema import validate_cloudconfig_schema
+-from cloudinit.lifecycle import log_with_downgradable_level
+ from cloudinit.reporting import events
+ from cloudinit.settings import (
+     PER_INSTANCE,
+@@ -49,8 +47,6 @@ from cloudinit.settings import (
+     CLOUD_CONFIG,
+ )
+ 
+-Reason = str
+-
+ # Welcome message template
+ WELCOME_MSG_TPL = (
+     "Cloud-init v. {version} running '{action}' at "
+@@ -316,104 +312,6 @@ def _should_bring_up_interfaces(init, ar
+         return False
+     return not args.local
+ 
+-
+-def _should_wait_via_user_data(
```

Comment:
Removing these function definitions means that we'll have to update the patch anytime we change this function, however if we just remove the callsite and leave it here that would accomplish the same without as much patch burden.

Thoughts?

Reply:
Good point, I'll update the patch.